preupg/xml_to_json.py

```diff
@@ -0,0 +1,404 @@
+
+        test_result_dict = self.parseDictOfElements(self.testResult, self.testResult_elements)
+        self.addOptToResultDict(self.result_dict, "general", test_result_dict)
+
+    def processResults(self):
+        """
+        Process every <rule-result> element
+        """
+        all_results = {}
+        for rule in self.rule_results:
+            rule_result_dict = {}
+            element_idref = self.getDataFromElement(rule, "idref")
+            #  possibility to process also <rule-result> attibutes
+            rule_result_dict[element_idref] = self.parseDictOfElements(rule, self.rule_result_elements)
+            all_results.update(rule_result_dict)
+
+        self.addOptToResultDict(self.result_dict, "results", all_results)
+
+    def parseDictOfElements(self, outer_element, elements_dict):
+        """
+            Parse text or attributes inside
+            outer_element into dictionary
+            based on element_dict
+
+        element_dict :
+           {result: text} -> text
+           {check: system} -> attribute system
+           {check: [system,text]} -> more than one attr/text
+                                    from the same element
+        """
+        element_dict = {}
+        for element_name, attribute in elements_dict.iteritems():
+            try:
+                element = self.findInnerElement(outer_element, element_name)
+            except MissingElementException as missing_ele_report:
+                log_message(missing_ele_report, level=logging.ERROR)
+
+            self.checkFoundElement(element, attribute)
+            if element:
+                try:
+                    element_dict.update(self.parseElementAttributes(element, attribute))
+                except EmptyTextException:
+                    pass
+                except AttrNotFoundException as err_mesg:
+                    log_message(err_mesg, level=logging.ERROR)
+
+        return element_dict
+
+    def checkFoundElement(self, element, attribute):
+        '''
+        Control each element whether it makes sense
+        to process them (like text is empty attribute
+        doesn't exit etc..)
+
+        '''
+        for elem in element:
+            # {check: [system,text]}
+            if type(attribute) is list:
+                for attr in attribute:
+                    try:
+                        self.checkForValidAttribute(elem, attr)
+                    except AttrNotFoundException as err_mesg:
+                        attribute.remove(attr)
+                        if self.verbose:
+                            log_message(err_mesg, level=logging.ERROR)
+            # {check: system} -> attribute system
+            else:
+                try:
+                    self.checkForValidAttribute(elem, attribute)
+                except AttrNotFoundException as err_mesg:
+                    element.remove(elem)
+                    if self.verbose:
+                        log_message(err_mesg, level=logging.ERROR)
```

Comment:
I'm getting the following traceback. It's because the function _log_message_ expects a string but you're passing an Exception object. You can fix it by:

_log_message(err_mesg.**message**, level=logging.ERROR)_
```
Traceback (most recent call last):
  File "./preupg-xml-to-json", line 30, in <module>
    main()
  File "./preupg-xml-to-json", line 27, in main
    json_file.write(convertor.convert())
  File "/home/mbocek/repos/rhel_pkgs/preupgrade-assistant-majo/preupg/xml_to_json.py", line 68, in convert
    self.processResults()
  File "/home/mbocek/repos/rhel_pkgs/preupgrade-assistant-majo/preupg/xml_to_json.py", line 148, in processResults
    rule_result_dict[element_idref] = self.parseDictOfElements(rule, self.rule_result_elements)
  File "/home/mbocek/repos/rhel_pkgs/preupgrade-assistant-majo/preupg/xml_to_json.py", line 172, in parseDictOfElements
    self.checkFoundElement(element, attribute)
  File "/home/mbocek/repos/rhel_pkgs/preupgrade-assistant-majo/preupg/xml_to_json.py", line 207, in checkFoundElement
    log_message(err_mesg, level=logging.ERROR)
  File "/home/mbocek/repos/rhel_pkgs/preupgrade-assistant-majo/preupg/logger.py", line 64, in log_message
    sys.stdout.write(message.encode(settings.defenc))
AttributeError: 'AttrNotFoundException' object has no attribute 'encode'
```